gui/guiComponent.py

```diff
@@ -0,0 +1,229 @@
+        frame.overrideredirect(True)
+        frame.option_add("*Font", "courier 12")
+        frame.withdraw()
+        self.frame = frame
+        self.hidden = True
+
+    def is_closed(self):
+        if self.closed:
+            return True
+        return False
+
+    def close(self):
+        if self.closed:
+            return
+        self.closed = True
+        self.frame.destroy()
+        self.frame.update()
+        self.frame = None
+
+    def reset(self):
+        for child in self.frame.winfo_children():
+            child.destroy()
+
+    def add_components(self):
+        pass
+
+    def create(self, x_cord, y_cord):
+        if not self.closed:
+            return
+        self.closed = False
+        self.prepare_window()
+        self.add_components()
+        windowToFront(self.frame)
+        self.frame.deiconify()
+        self.frame.geometry(f"+{x_cord}+{y_cord}")
+        self.frame.update()
+        self.opened = time.time()
+
+    def show(x_cord, y_cord):
+        self.hidden = False
+        windowToFront(self.frame)
+        self.frame.deiconify()
+        self.frame.geometry(f"+{x_cord}+{y_cord}")
+        self.frame.update()
+        self.opened = time.time()
+
+    def hide(self, refocus=True):
+        if self.hidden:
+            return
+        self.hidden = True
+        self.frame.withdraw()
+        self.frame.update()
+        if refocus:
+            windowRefocus("path of exile")
+
+    def create_at_cursor(self):
+        if not self.closed:
+            return
+        self.closed = False
+        self.prepare_window()
+        self.add_components()
+        self.show_at_cursor()
+
+    def show_at_cursor(self):
+        self.hidden = False
+        windowToFront(self.frame)
+        self.frame.update()
+        m_x = self.frame.winfo_pointerx()
+        m_y = self.frame.winfo_pointery()
+        def get_monitor_from_coord(x, y):
+            monitors = screeninfo.get_monitors()
+            for m in reversed(monitors):
+                if m.x <= x <= m.width + m.x and m.y <= y <= m.height + m.y:
+                    return m
+            return monitors[0]
+
+        # Get the screen which contains top
+        current_screen = get_monitor_from_coord(self.frame.winfo_x(), self.frame.winfo_y())
+        # Get the monitor's size
+        root_w = self.frame.winfo_width()
+        root_h = self.frame.winfo_height()
+
+        if m_x + root_w >= current_screen.width:
+            m_x = current_screen.width - root_w - 5
+
+        if m_y + root_h >= current_screen.height:
+            m_y = current_screen.height - root_h - 5
+
+        self.frame.deiconify()
+        self.frame.geometry(f"+{m_x}+{m_y}")
+        self.frame.update()
+        self.opened = time.time()
+
+class GuiRunningComponent(GuiComponent):
+    def stop(self):
+        self.frame.quit()
+        self.close()
+
+    def run(self):
+        self.frame.mainloop()
+    
+    def prepare_window(self):
+        tk = Tk().withdraw()
```

Comment:
This should be removed as well, it is exactly same issue as in the other component

Reply:
Yeah missed that one